### PR TITLE
chore(CI): migrate Travis CI ⇒ GitHub Actions

### PR DESCRIPTION
**Pull request by vytas7**
We are getting Windows and Mac support with zero effort.
And runtimes are really quick, around 6-7 minutes for the whole suite.

Things to do:
- [x] Update `on` hooks to reflect Falcon workflow
- [x] Codecov integration
- [x] CPython 3.5.2 gate (easiest to probably just `apt-get` on 16.04)
- [x] Update `README.rst` banners
- [x] Add a big endian architecture (could be done in QEMU, @CaselIT has an ARM recipe from a SQLAlchemy PR); had to do this in a separate manual action unfortunately, taking ~40 minutes. There is optimization potential, but bringing that to the vicinity of 6-7 minutes does not sound realistic at this point.

A workflow run example on my fork:
https://github.com/vytas7/falcon/actions/runs/198951620

### CODE REVIEW

**Comment by CaselIT** [comment]
Question: do we want to have also arm tests?

**Comment by vytas7** [comment]
No, I don't think that is a high priority tbh.
But maybe we could later build a separate workflow for building wheels, and that should probably include ARM (w/e architecture Ubuntu and Amazon are using, it's called just `arm64` I think).

**Comment by CaselIT** [comment]
For building wheels you can use the manylinux image and or this image to automate it
- https://github.com/pypa/manylinux
- https://github.com/RalfG/python-wheels-manylinux-build

**Comment by vytas7** [comment]
Which architectures does that action do? Is `s390x` included, which ARM flavours are included?

**Comment by CaselIT** [comment]
I think all of the supported manylinux images. https://github.com/RalfG/python-wheels-manylinux-build/tags
Manylinux 2014 does support s390x: https://github.com/pypa/manylinux#manylinux2014
When creating wheels for sqlalchemy, I've read that is still suggested to use manylinux2010 (or manylinux1 if support for older linux is required). I'm using the 2010 in sqlalchemy, but it manages to automatically create also the manylinux1 version. Both are uploaded and pip chooses the appropriate one for the installed system

**Comment by vytas7** [comment]
Aha, that's awesome. We'll probably have to skip 32 bit architectures though, because we don't test enough on them. For instance, WSGI cythonized buffered reader is disabled by design on 32-bit, falling back to pure Python.

**Comment by CaselIT** [comment]
32bit is really only windows at this point.

**Comment by vytas7** [comment]
`i686` is well 32 bit?

**Comment by CaselIT** [comment]
Yes, but who is still using 32bit linux at this point?

**Comment by vytas7** [comment]
True; that's also why I'm saying we should drop them. All major distributions have now dropped 32 bits, or demoted that to community support.

**Comment by vytas7** [comment]
And thanks @CaselIT for your insights, that's really useful.

I _think_ we may decide to cut straight to `manylinux2014` ([PEP 599]()), since we dropped Python 2, <= Python 3.4 support, so you can probably expect people running Python 3.5+ on `manylinux2014` compatible platforms.

**Comment by CaselIT** [comment]
The only problem may be pip. Note that if it works like in sqlalchemy, 2014 may also generate 2010 and 1 wheels.
Pip can then download the appropriate one

**Comment by CaselIT** [comment]
Following up from https://github.com/falconry/falcon/issues/1281. I think there should also be a job that test that the readme is accepted by pypi using `twine check`, since I don't think that the create wheel action is the best place to do that check.

Incidentally the current readme in master fails the test :)

**Comment by vytas7** [comment]
Agreed; to that end, we ought to add a `tox` environment for that.

File: tox.ini [hunk 1]
```diff
@@ -276,6 +283,20 @@ basepython = python3.8
 deps =
 commands = {toxinidir}/tools/check-vendored.sh
 
+# --------------------------------------------------------------------
+# Package sanity check with twine
+# --------------------------------------------------------------------
+
+[testenv:twine_check]
```
Comment (kgriffs) [inline tox.ini:290]: +1

**Pull request merged**